libcasr/src/csharp.rs

```diff
@@ -0,0 +1,128 @@
+//! C Sharp module implements `ParseStacktrace` and `Exception` traits for C Sharp reports.
+use crate::exception::Exception;
+use crate::stacktrace::{ParseStacktrace, Stacktrace};
+
+use crate::error::*;
+use crate::execution_class::ExecutionClass;
+use crate::stacktrace::StacktraceEntry;
+
+use regex::Regex;
+
+const PARSE_STACK_FRAME_ERR_PREFIX: &str = "Couldn't parse stacktrace line: ";
+
+/// Structure provides an interface for processing the stack trace.
+pub struct CSharpStacktrace;
+
+impl CSharpStacktrace {
+    fn parse_stacktrace_entry(entry: &str, use_mono_format: bool) -> Result<StacktraceEntry> {
```

Comment:
Replace bool parameter with enum value. Create an enum containing Mono and dotNET values (for readability).